src/Uno.UI/UI/Xaml/Controls/Image/Image.wasm.cs

```diff
@@ -254,94 +259,96 @@ protected override Size MeasureOverride(Size availableSize)
 
 		protected override Size ArrangeOverride(Size finalSize)
 		{
-			(double x, double y, double? width, double? height) getHtmlImagePosition()
-			{
-				var sourceRect = new Windows.Foundation.Rect(Windows.Foundation.Point.Zero, _lastMeasuredSize);
-				var imageRect = new Windows.Foundation.Rect(Windows.Foundation.Point.Zero, finalSize);
-
-				this.MeasureSource(imageRect, ref sourceRect);
-				this.ArrangeSource(imageRect, ref sourceRect);
-
-				switch (Stretch)
-				{
-					case Stretch.Fill:
-					case Stretch.None:
-						return (sourceRect.X, sourceRect.Y, sourceRect.Width, sourceRect.Height);
-
-					case Stretch.Uniform:
-						if (finalSize.Width >= _lastMeasuredSize.Width)
-						{
-							// /---/-----\---\
-							// |   |     |   |
-							// |   |     |   |
-							// |   |     |   |
-							// |   |     |   |
-							// |   |     |   |
-							// \---\-----/---/
-
-							return (sourceRect.X, sourceRect.Y, null, sourceRect.Height);
-						}
-						else
-						{
-							return (sourceRect.X, sourceRect.Y, sourceRect.Width, null);
-						}
-
-					case Stretch.UniformToFill:
-						var adjustedSize = this.AdjustSize(finalSize, _lastMeasuredSize);
-
-						if (adjustedSize.Height <= finalSize.Height)
-						{
-							//
-							// /-------------\
-							// |             |
-							// /-------------\
-							// |             |
-							// |             |
-							// |             |
-							// |             |
-							// \-------------/
-							// |             |
-							// \-------------/
-							//
-
-							return (sourceRect.X, sourceRect.Y, null, finalSize.Height);
-						}
-						else
-						{
-							return (sourceRect.X, sourceRect.Y, finalSize.Width, null);
-						}
-
-					default:
-						throw new NotSupportedException();
-				}
-			}
-
-			var position = getHtmlImagePosition();
+			var sourceRect = new Windows.Foundation.Rect(Windows.Foundation.Point.Zero, _lastMeasuredSize);
+			this.MeasureSource(finalSize, ref sourceRect);
+			this.ArrangeSource(finalSize, ref sourceRect);
+
+			//(double? width, double? height) getHtmlImagePosition()
```

Comment:
Moved to common helper ?

Reply:
This logic is already covered by the `ArrangeSource` helper.  But there's still commented code to remove...